boot/bootutil/src/loader.c

```diff
@@ -1335,36 +1354,42 @@ boot_update_security_counter(uint8_t image_index, int slot,
 #define SEC_SLOT_TOUCHED 1
 #define SEC_SLOT_ASSIGNED 2
 
-#if (MCUBOOT_IMAGE_NUMBER == 2) && defined(PM_B0_ADDRESS) && \
-      !defined(CONFIG_NRF53_MULTI_IMAGE_UPDATE)
-/* This configuration is peculiar - the one physical secondary slot is
- * mocking two logical secondary
- */
-#define SEC_SLOT_PHYSICAL_CNT 1
-#else
-#define SEC_SLOT_PHYSICAL_CNT MCUBOOT_IMAGE_NUMBER
-#endif
-
-static uint8_t sec_slot_assignmnet[SEC_SLOT_PHYSICAL_CNT] = {0};
+static uint8_t sec_slot_assignmnet[MCUBOOT_IMAGE_NUMBER] = {0};
 
 static inline void sec_slot_touch(struct boot_loader_state *state)
 {
-    uint8_t idx = (SEC_SLOT_PHYSICAL_CNT == 1) ? 0 : BOOT_CURR_IMG(state);
+#if CONFIG_MCUBOOT_MCUBOOT_IMAGE_NUMBER != -1
```

Comment:
now sec_slot_assignment can be marked twice. I do not understand that.

Reply:
This feature is completely broken in child/parent image as I've said many time, depending on what you have, you can get MCUboot to mark an image for upgrade, after which the cleanup code will run, then it will upgrade the primary for 4 bytes, and bricks the module. And brick meaning if access port protection is enabled, the device is irrecoverably dead, likely out in the field. This code here actually works by checking the slots properly and marking both as used if it is a shared slot. Note that all of these test cases are checked as part of sdk-mcuboot CI and pass